SofaKernel/modules/Sofa.Topology/src/sofa/topology/Element.h

```diff
@@ -0,0 +1,52 @@
+/******************************************************************************
+*                 SOFA, Simulation Open-Framework Architecture                *
+*                    (c) 2006 INRIA, USTL, UJF, CNRS, MGH                     *
+*                                                                             *
+* This program is free software; you can redistribute it and/or modify it     *
+* under the terms of the GNU Lesser General Public License as published by    *
+* the Free Software Foundation; either version 2.1 of the License, or (at     *
+* your option) any later version.                                             *
+*                                                                             *
+* This program is distributed in the hope that it will be useful, but WITHOUT *
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or       *
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License *
+* for more details.                                                           *
+*                                                                             *
+* You should have received a copy of the GNU Lesser General Public License    *
+* along with this program. If not, see <http://www.gnu.org/licenses/>.        *
+*******************************************************************************
+* Authors: The SOFA Team and external contributors (see Authors.txt)          *
+*                                                                             *
+* Contact information: contact@sofa-framework.org                             *
+******************************************************************************/
+#pragma once
+
+#include <sofa/topology/config.h>
+
+#include <sofa/type/stdtype/fixed_array.h>
+#include <sofa/topology/Point.h>
+
+#include <type_traits>
+
+namespace sofa::topology
+{
+
+template <typename GeometryElement>
+struct Element : public sofa::type::stdtype::fixed_array<sofa::Index, GeometryElement::NumberOfNodes>
+{
+    constexpr Element() noexcept
+    {
+        std::fill(this->begin(), this->end(), sofa::InvalidID);
+    }
+
+    template< typename... ArgsT
```

Comment:
Here you could use the power of C++17 [fold expression](https://en.cppreference.com/w/cpp/language/fold)

```c++
template <
    typename... ArgsT, 
    typename = std::enable_if_t <
        (GeometryElement::NumberOfNodes == sizeof...(ArgsT)) && 
        (std::is_convertible_v<ArgsT,  sofa::Index> && ...)
    >
>
constexpr Element(ArgsT&&... args) noexcept
    : sofa::type::stdtype::fixed_array< sofa::Index, GeometryElement::NumberOfNodes >
{ static_cast<sofa::Index>(std::forward< ArgsT >(args))... }
{}
```

Or, maybe a [static_assert](https://en.cppreference.com/w/cpp/language/static_assert) would be more appropriate?

```c++
template <
    typename... ArgsT, 
    typename = std::enable_if_t < (std::is_convertible_v<ArgsT,  sofa::Index> && ...)>
>
constexpr Element(ArgsT&&... args) noexcept
    : sofa::type::stdtype::fixed_array< sofa::Index, GeometryElement::NumberOfNodes >
{ static_cast<sofa::Index>(std::forward< ArgsT >(args))... }
{
    static_assert(GeometryElement::NumberOfNodes == sizeof...(ArgsT), "Trying to construct the element with an incorrect number of nodes.");
}
```

Reply:
the static_assert version is interesting indeed. 
lets discuss while doing the global review